utils/health/build-scripts/clang-build-time-analyzer-build.sh

```diff
@@ -0,0 +1,54 @@
+#!/bin/bash -e
+
+# Copyright (c) 2014-2020, The Monero Project
+#
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without modification, are
+# permitted provided that the following conditions are met:
+#
+# 1. Redistributions of source code must retain the above copyright notice, this list of
+#    conditions and the following disclaimer.
+#
+# 2. Redistributions in binary form must reproduce the above copyright notice, this list
+#    of conditions and the following disclaimer in the documentation and/or other
+#    materials provided with the distribution.
+#
+# 3. Neither the name of the copyright holder nor the names of its contributors may be
+#    used to endorse or promote products derived from this software without specific
+#    prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY
+# EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF
+# MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL
+# THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+# SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+# PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+# STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF
+# THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+#
+# Parts of this file are originally copyright (c) 2012-2013 The Cryptonote developers
+
+# This script checkouts and builds ClangBuildAnalyzer.
+# The result is put into bin directory
+
+DIR_THIS="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"
+
+TAG=v1.2.0
+PROG=ClangBuildAnalyzer
+DIR_OUT=$DIR_THIS/../bin
+
+cd /tmp
+if [ ! -d $PROG ]; then
+	git clone https://github.com/aras-p/$PROG.git
+fi
```

Comment:
It's really not clear from the script name that this actually gets some code from an external souce and executes stuff from it. Is this program not in common package repos ?

Reply:
It is not.
What do you propose to do? Keep a checked out source in Monero project? Set it as a submodule, even if it's optional? 
How would you name the script?

Reply:
If people want to use it, they download it. I really hate when build rules fetch stuff and run it. Submodule would be definitely better. At least you checkout a particular commit hash, but even then it feels iffy. Anyway, maybe others are less weirded out by this.

Reply:
Maybe just add CMake support and document all the steps required in `README.md` and we are good?

Reply:
Ah, or you could implement a Dockerfile instead. I would prefer this tbh.

Reply:
> Maybe just add CMake support and document all the steps required in `README.md` and we are good?

I think I will go this way, since I have 0 knowledge about Docker as of now. 
I didn't quire understand what you meant by CMake support in this particular case.

Reply:
Having the script and making it very clear it will download/run stuff would be OK I think.